src/ui/components/AppWrapper/AppWrapper.tsx

```diff
@@ -118,11 +118,53 @@ const AppWrapper = (props: { children: ReactNode }) => {
   const authentication = useAppSelector(getAuthentication);
   const [initialised, setInitialised] = useState(false);
   const [agentInitErr, setAgentInitErr] = useState(false);
+  const [isOffline, setIsOffline] = useState(true);
 
   useEffect(() => {
     initApp();
   }, []);
 
+  useEffect(() => {
+    if (!isOffline && initialised) {
+      const handleMessages = async () => {
+        AriesAgent.agent.connections.onConnectionKeriStateChanged((event) => {
+          return connectionKeriStateChangedHandler(event, dispatch);
+        });
+        AriesAgent.agent.signifyNotifications.onNotificationKeriStateChanged(
+          (event) => {
+            return keriNotificationsChangeHandler(event, dispatch);
+          }
+        );
+        AriesAgent.agent.credentials.onAcdcKeriStateChanged((event) => {
+          return keriAcdcChangeHandler(event, dispatch);
+        });
+
+        const oldMessages = (
+          await Promise.all([
+            AriesAgent.agent.credentials.getKeriCredentialNotifications(),
+            AriesAgent.agent.identifiers.getUnhandledMultisigIdentifiers({
+              isDismissed: false,
+            }),
+          ])
+        )
+          .flat()
+          .sort(function (messageA, messageB) {
+            return messageA.createdAt.valueOf() - messageB.createdAt.valueOf();
+          });
+        oldMessages.forEach(async (message) => {
+          await keriNotificationsChangeHandler(message, dispatch);
+        });
+        // Fetch and sync the identifiers, contacts and ACDCs from KERIA to our storage
+        await Promise.all([
+          AriesAgent.agent.identifiers.syncKeriaIdentifiers(),
+          AriesAgent.agent.connections.syncKeriaContacts(),
+          AriesAgent.agent.credentials.syncACDCs(),
```

Comment:
I think we should just call these, and the collection of the `oldMessages` in this useEffect. So we can remove it from the old area (bottom of `initApp`)